### PR TITLE
Fix spaced delegate

### PR DESCRIPTION
**Pull request by TheOneRing**

### CODE REVIEW

File: src/gui/spaces/spacesdelegate.cpp [hunk 1]
```diff
@@ -169,7 +166,8 @@ QStyleOptionButton SpacesDelegate::openBrowserButtonRect(const QStyleOptionViewI
     opt.iconSize = QSize { px, px };
 
     opt.rect.setSize(QApplication::style()->sizeFromContents(
-        QStyle::CT_PushButton, &opt, opt.fontMetrics.size(Qt::TextSingleLine, opt.text)));
+        QStyle::CT_PushButton, &opt, opt.fontMetrics.size(Qt::TextSingleLine, opt.text), option.widget));
+    opt.rect.setSize(opt.rect.size() + QSize { px * 2, 0 });
```
Comment (TheOneRing) [inline src/gui/spaces/spacesdelegate.cpp:170]: This is just a hack and can't be right.
There needs to be a better solution.

**Comment by ownclouders** [comment]
Results for <strong>GUI-tests</strong> https://drone.owncloud.com/owncloud/client/13234/6/1

:boom: The GUI tests failed.

GUI Logs: https://cache.owncloud.com/public/owncloud/client/13234/guiReportUpload/index.html

<details>

Server Logs: https://cache.owncloud.com/public/owncloud/client/13234/guiReportUpload/serverlog.log

Screenshots:
- [Filenames_that_are_rejected_by_the_server_are_reported.png](https://cache.owncloud.com/public/owncloud/client/13234/guiReportUpload/screenshots/Filenames_that_are_rejected_by_the_server_are_reported.png)

</details>

File: src/gui/spaces/spacesdelegate.cpp (hunk 1 above)
Comment (fmoc) [reply src/gui/spaces/spacesdelegate.cpp:170]: Why not leave a TODO in the code?

**Review by fmoc — CHANGES_REQUESTED** [review]
See my previous comment.

File: src/gui/spaces/spacesdelegate.cpp (hunk 1 above)
Comment (TheOneRing) [reply src/gui/spaces/spacesdelegate.cpp:170]: I hoped that we could find a solution in the reviews.

Comment (fmoc) [reply src/gui/spaces/spacesdelegate.cpp:170]:
```suggestion
    opt.rect.setSize(opt.rect.size() + QSize { px + QApplication::style()->pixelMetric(QStyle::PM_ButtonMargin), 0 });
```

Works pretty well apparently, the button looks like other push buttons with icons:

screenshot_2022-10-04_16-52-49

**Review by fmoc — APPROVED** [review]

**Pull request merged**